pkg/products/marin3r/rateLimitService.go

```diff
@@ -550,6 +550,11 @@ func (r *RateLimitServiceReconciler) ensureLimits(ctx context.Context, client k8
 		return integreatlyv1alpha1.PhaseAwaitingComponents, nil
 	}
 
+	for _, pod := range rateLimitPods.Items {
+		if pod.Status.Phase == corev1.PodFailed {
+			return integreatlyv1alpha1.PhaseFailed, fmt.Errorf("pod failed to start")
```

Comment:
Following the verification steps, it seems the following error is shown:
```
    product reconcilation errors : failed installation of marin3r: Failed
    executing command [/bin/sh -c wget -qO -
    http://127.0.0.1:8080/limits/apicast-ratelimit] on
    redhat-rhoam-marin3r/ratelimit-57d9d64d74-fwzfd: unable to upgrade
    connection: container not found ("ratelimit")
```

Seems like the Pods with an incorrect image goes into a pending state instead which was why this part of the code was not reached
![image](https://user-images.githubusercontent.com/24636860/173618265-03e3b23b-158d-4bea-bc40-b892d94e1369.png)

So maybe we want to do a check for any pods not in a running state instead :thinking: 

Something like:

```suggestion
	for _, pod := range rateLimitPods.Items {
		if pod.Status.Phase != corev1.PodRunning {
			return integreatlyv1alpha1.PhaseInProgress, fmt.Errorf("waiting for rate limit pods to be ready")
```